src/Timeline.cpp

```diff
@@ -73,6 +75,107 @@ Timeline::Timeline(int width, int height, Fraction fps, int sample_rate, int cha
 	final_cache->SetMaxBytesFromInfo(OPEN_MP_NUM_PROCESSORS * 2, info.width, info.height, info.sample_rate, info.channels);
 }
 
+// Constructor for the timeline (which loads a JSON structure from a file path, and initializes a timeline)
+Timeline::Timeline(std::string projectPath, bool convert_absolute_paths) :
+		is_open(false), auto_map_clips(true), managed_cache(true), path(projectPath) {
+
+	// Create CrashHandler and Attach (incase of errors)
+	CrashHandler::Instance();
+
+	// Init final cache as NULL (will be created after loading json)
+	final_cache = NULL;
+
+	// Init viewport size (curve based, because it can be animated)
+	viewport_scale = Keyframe(100.0);
+	viewport_x = Keyframe(0.0);
+	viewport_y = Keyframe(0.0);
+
+	// Init background color
+	color.red = Keyframe(0.0);
+	color.green = Keyframe(0.0);
+	color.blue = Keyframe(0.0);
+
+	// Check if path exists
+	QFileInfo filePath(QString::fromStdString(path));
+	if (!filePath.exists()) {
+		throw InvalidFile("File could not be opened.", path);
+	}
+
+	// Check OpenShot Install Path exists
+	Settings *s = Settings::Instance();
+	QDir openshotPath(QString::fromStdString(s->PATH_OPENSHOT_INSTALL));
+	if (!openshotPath.exists()) {
+		throw InvalidFile("PATH_OPENSHOT_INSTALL could not be found.", s->PATH_OPENSHOT_INSTALL);
+	}
+	QDir openshotTransPath(openshotPath.filePath("transitions"));
+	if (!openshotTransPath.exists()) {
+		throw InvalidFile("PATH_OPENSHOT_INSTALL/transitions could not be found.", openshotTransPath.path().toStdString());
+	}
+
+	// Determine asset path
+	QString asset_name = filePath.baseName().left(30) + "_assets";
+	QDir asset_folder(filePath.dir().filePath(asset_name));
+	if (!asset_folder.exists()) {
+		// Create directory if needed
+		asset_folder.makeAbsolute();
+	}
+
+	// Load UTF-8 project file into QString
+	QFile projectFile(QString::fromStdString(path));
+	projectFile.open(QFile::ReadOnly);
+	QString projectContents = QString::fromUtf8(projectFile.readAll());
+
+	// Convert all relative paths into absolute paths (does not support relative paths with ../)
+	// In otherwords, assets and files must be located in a child/sub-folder (and not from outside this folder)
+	if (convert_absolute_paths) {
+		// Convert all paths into absolute (if requested)
+		QRegularExpression pathRegex(QStringLiteral("\"(image|path)\":.*?\"(?:\\./)?(?!@assets|@transitions+)(.*?)\""));
```

Comment:
OK, first off, this is a horror show. :laughing: C++11 supports raw string literals, same as Python (but with a slightly different format)
```suggestion
		QRegularExpression pathRegex(QStringLiteral(
			R"__(
				"(image|path)":.*?"(?:\./)?(?!@assets|@transitions+)(.*?)
			")__", QRegularExpression::ExtendedPatternSyntaxOption
		));
```
...Much more readable. And if I plug that into [Debuggex (live session link)](https://www.debuggex.com/r/uFO61iT7iS165S-E
), a couple of parts are iffy — `@transitions+` is the major red flag, since it would match `@transitions`, `@transitionss`, @transitionssssssss`... unlikely to be the intent.

I think this is more like what you want:
```c++
R"__(
	"(image|path)":\s*?"(?:\./)?(?!@(?:assets|transitions)/)(.*)"
)__"
```
...However, that's still assuming QRegularExpression supports all of the advanced features in use, including negative lookahead — RE implementations can be surprisingly lacking. Still trying to confirm that in the documentation.